### PR TITLE
feat(curriculum): Add play scene keyboard shortcut

### LINKED ISSUE #52696: Add some keyboard shortcuts to A2 English for Developers
### Is your feature request related to a problem? Please describe.

NA

### Describe the solution you'd like

I wish if there's a hotkey, like Space to run the new video, sometimes Ctrl+enter does not submit the input.
Anything in this direction would be more helpful.

### Describe alternatives you've considered

NA

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by abhinavkgrd**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #52696

Adds "ctrl + space" keyboard shortcut for playing scene as suggested in https://github.com/freeCodeCamp/freeCodeCamp/issues/52696#issuecomment-1884048129

Also updates the shortcut modal with the new shortcut info 

<details>
<summary > Screenshot attached </summary>

Screenshot 2024-03-24 at 4 20 35 PM</details>

### CODE REVIEW

**Comment by gikf** [comment]
> There is some issue with the translation, I have added the string to the english translation file, but its not getting translated and the key is directly printed

Running `pnpm run clean:client` should fix that.

**Comment by abhinavkgrd** [comment]
Just checking up, is anything blocking the PR?

File: client/src/templates/Challenges/components/scene/scene.tsx [hunk 1]
```diff
@@ -8,7 +8,15 @@ import Character from './character';
...
 
-export function Scene({ scene }: { scene: FullScene }): JSX.Element {
+export function Scene({
+  scene,
+  isPlaying,
+  setIsPlaying
+}: {
+  scene: FullScene;
+  isPlaying: boolean;
+  setIsPlaying: (isPlaying: boolean) => void;
```
Comment (naomi-lgbt) [inline client/src/templates/Challenges/components/scene/scene.tsx:18]:
```suggestion
  setIsPlaying: (shouldPlay: boolean) => void;
```

File: client/src/templates/Challenges/dialogue/show.tsx [hunk 2]
```diff
@@ -171,6 +174,12 @@ class ShowDialogue extends Component<ShowDialogueProps, ShowDialogueState> {
     });
   };
 
+  setIsScenePlaying = (isPlaying: boolean) => {
+    this.setState({
+      isScenePlaying: isPlaying
+    });
+  };
```
Comment (naomi-lgbt) [inline client/src/templates/Challenges/dialogue/show.tsx:181]:
```suggestion
  setIsScenePlaying = (shouldPlay: boolean) => {
    this.setState({
      isScenePlaying: shouldPlay
    });
  };
```

File: client/src/templates/Challenges/fill-in-the-blank/show.tsx [hunk 3]
```diff
@@ -243,6 +246,12 @@ class ShowFillInTheBlank extends Component<
     return '';
   }
 
+  setIsScenePlaying = (isPlaying: boolean) => {
+    this.setState({
+      isScenePlaying: isPlaying
+    });
+  };
```
Comment (naomi-lgbt) [inline client/src/templates/Challenges/fill-in-the-blank/show.tsx:253]:
```suggestion
  setIsScenePlaying = (shouldPlay: boolean) => {
    this.setState({
      isScenePlaying: shouldPlay
    });
  };
```

File: client/src/templates/Challenges/odin/show.tsx [hunk 4]
```diff
@@ -204,6 +207,12 @@ class ShowOdin extends Component<ShowOdinProps, ShowOdinState> {
     });
   };
 
+  setIsScenePlaying = (isPlaying: boolean) => {
+    this.setState({
+      isScenePlaying: isPlaying
+    });
+  };
```
Comment (naomi-lgbt) [inline client/src/templates/Challenges/odin/show.tsx:214]:
```suggestion
  setIsScenePlaying = (shouldPlay: boolean) => {
    this.setState({
      isScenePlaying: shouldPlay
    });
  };
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Thanks for your efforts here! This is a bit of a tricky one, so...

Operating under the assumption that we **only** want `Ctrl+Space` to start the audio, and **not** to be able to stop it:

- First, `isPlaying` isn't an accurate variable name here, because that's not actually what we're passing to the `setIsPlaying` functions. We're passing a boolean representing whether it **should be playing** or not. Review comments propose a clearer variable name.
- Second, we're missing the `playScene` prop in the `Hotkeys` elements for both the fill in the blank layout and the odin layout.

**Review by naomi-lgbt — APPROVED** [review]
I think this is fine as is. I played around with making it toggle start/stop, and restarting the dialogue a second time resulted in strange behaviour. Fixing that would be out of scope for this PR, so I'm happy to land as is. :3

File: client/i18n/locales/english/translations.json [hunk 5]
```diff
@@ -1037,7 +1037,8 @@
     "focus-editor": "Focus Editor",
     "focus-instructions-panel": "Focus Instructions Panel",
     "navigate-previous": "Navigate Previous Exercise",
-    "navigate-next": "Navigate Next Exercise"
+    "navigate-next": "Navigate Next Exercise",
```
Comment (Sembauke) [inline client/i18n/locales/english/translations.json:1040]:
```suggestion
    "navigate-next": "Navigate To Next Exercise",
```

File: client/i18n/locales/english/translations.json [hunk 6]
```diff
@@ -1037,7 +1037,8 @@
     "focus-editor": "Focus Editor",
     "focus-instructions-panel": "Focus Instructions Panel",
     "navigate-previous": "Navigate Previous Exercise",
```
Comment (Sembauke) [inline client/i18n/locales/english/translations.json:1039]:
```suggestion
    "navigate-previous": "Navigate To Previous Exercise",
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

**Comment by abhinavkgrd** [comment]
Heyy @Sembauke , They are not in scope for this PR , Maybe it would be better to fix them in different PR? But, yes I can include them if you want.

**Review by Sembauke — APPROVED** [review]

**Pull request merged**